src/en/onepunchmanonline/src/eu/kanade/tachiyomi/extension/en/onepunchmanonline/OnePunchManOnline.kt

```diff
@@ -0,0 +1,116 @@
+    override fun fetchPopularManga(page: Int): Observable<MangasPage> {
+        return Observable.just(MangasPage(listOf(createManga()), false))
+    }
+
+    override fun popularMangaRequest(page: Int): Request = throw UnsupportedOperationException()
+    override fun popularMangaSelector(): String = throw UnsupportedOperationException()
+    override fun popularMangaFromElement(element: Element): SManga = throw UnsupportedOperationException()
+    override fun popularMangaNextPageSelector(): String? = null
+
+    // =========================================================================
+    //  Latest Updates
+    // =========================================================================
+    override fun fetchLatestUpdates(page: Int) = fetchPopularManga(page)
+    override fun latestUpdatesRequest(page: Int) = throw UnsupportedOperationException()
+    override fun latestUpdatesSelector() = throw UnsupportedOperationException()
+    override fun latestUpdatesFromElement(element: Element) = throw UnsupportedOperationException()
+    override fun latestUpdatesNextPageSelector() = null
+
+    // =========================================================================
+    //  Search (Client-Side Safe Filtering)
+    // =========================================================================
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
+        return Observable.fromCallable {
+            val manga = createManga()
+            // If query matches title, return the manga. Otherwise return empty list.
+            if (query.isBlank() || manga.title.contains(query, ignoreCase = true)) {
+                MangasPage(listOf(manga), false)
+            } else {
+                MangasPage(emptyList(), false)
+            }
+        }
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList) = throw UnsupportedOperationException()
+    override fun searchMangaParse(response: Response) = throw UnsupportedOperationException()
+    override fun searchMangaSelector() = throw UnsupportedOperationException()
+    override fun searchMangaFromElement(element: Element) = throw UnsupportedOperationException()
+    override fun searchMangaNextPageSelector() = null
+
+    // =========================================================================
+    //  Manga Details
+    // =========================================================================
+    override fun fetchMangaDetails(manga: SManga): Observable<SManga> {
+        return Observable.just(createManga().apply { initialized = true })
+    }
+
+    override fun mangaDetailsParse(document: Document) = throw UnsupportedOperationException()
+
+    private fun createManga(): SManga {
+        return SManga.create().apply {
+            title = "One Punch Man"
+            url = "/"
+            thumbnail_url = "https://upload.wikimedia.org/wikipedia/en/c/c3/OnePunchMan_manga_cover.png"
```

Comment:
Highest resolution I could find: https://comicvine.gamespot.com/a/uploads/original/11111/111114608/3458824-8589005300-YICcI.jpg

Wikimedia might be more reliable, so its mostly up to you.